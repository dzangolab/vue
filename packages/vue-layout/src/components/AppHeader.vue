<template>
  <header>
    <slot name="logo">
      <Logo class="logo" />
    </slot>
    <slot name="menu">
      <MainMenu class="mainMenu" :class="{ menuExpanded: expanded }" />
    </slot>
    <slot name="userMenu"> </slot>
    <slot name="locales">
      <LocaleSwitcher class="locales" :class="{ menuExpanded: expanded }" />
    </slot>
    <slot name="toggle">
      <div class="toggle" @click="expanded = !expanded">
        <Icon v-if="expanded" icon="fa6-solid:bars-staggered" height="1.5rem" />
        <Icon v-else icon="fa6-solid:bars" height="1.5rem" />
      </div>
    </slot>
  </header>
</template>

<script lang="ts">
export default {
  data() {
    return {
      expanded: false,
    };
  },

  name: "AppHeader",
};
</script>

<script setup lang="ts">
import { LocaleSwitcher } from "@dzangolab/vue3-i18n";
import { Icon } from "@iconify/vue";

import Logo from "./Logo.vue";
import MainMenu from "./MainMenu.vue";
</script>

<style lang="css">
header {
  align-items: start;
  border-bottom: 1px solid var(--header-border-color, #4169e1);
  display: grid;
  font-size: var(--header-font-size, 1rem);
  grid-template-areas:
    "logo toggle"
    "mainMenu mainMenu"
    "userMenu userMenu"
    "locales locales";
  grid-template-columns: auto 1fr;
  grid-template-rows: auto auto;
  padding-bottom: var(--header-padding-top, 1em);
  padding-top: var(--header-padding-bottom, 1em);
}

header::before {
  content: "";
  background-color: var(--header-bg-color, lightblue);
  box-shadow: 0 0 0 100vmax var(--header-bg-color, lightblue);
  clip-path: inset(
    calc(-1 * var(--header-padding-top, 1em)) -100vmax calc(
        -1 * var(--header-padding-bottom, 1em)
      ) -100vmax
  );
  grid-column: 1 / end;
  grid-row: 1 / 1;
  height: 100%;
  z-index: -1;
}

header > nav {
  display: none;
  margin-top: 1rem;
}

header > .logo {
  width: 6rem;
  grid-area: logo;
  max-width: 200px;
}

header > .toggle {
  grid-area: toggle;
  justify-self: right;
  align-self: center;
  cursor: pointer;
}

header > .mainMenu {
  grid-area: mainMenu;
}

header > .userMenu {
  grid-area: userMenu;
}

header > .locales {
  grid-area: locales;
  width: 100%;
}

header .menuExpanded {
  display: flex;
  column-gap: var(--header-gap, 1rem);
}

@media screen and (min-width: 576px) {
  header {
    grid-template-areas: "logo mainMenu userMenu locales";
    align-items: center;
    grid-gap: var(--header-gap, 1rem);
    grid-template-columns: var(
      --header-grid-template-columns,
      auto 1fr auto 6rem
    );
    grid-template-rows: auto;
  }

  header > nav {
    display: flex;
    overflow: visible;
    margin-top: 0;
  }

  header > .toggle {
    display: none;
  }
}
</style>
