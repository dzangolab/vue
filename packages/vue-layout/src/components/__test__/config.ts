import type { AppConfig } from "@dzangolab/vue3-config";
import type { LocaleMessages, VueMessageType } from "@dzangolab/vue3-i18n";

const config: AppConfig = {
  apiBaseUrl: "http://localhost",
  appTitle: "Test app",
  appVersion: "0.1.0",
  copyright: {
    holder: "Dzango Technologies Limited",
    url: "https://dzango.com",
  },
  i18n: {
    fallbackLocale: "en",
    messages: {} as LocaleMessages<VueMessageType>,
    supportedLocales: ["en", "fr"],
  },
  features: {
    showVersion: true,
    sentry: true,
  },
  layout: {
    mainMenu: [
      {
        name: "Home",
        route: "home",
      },
      {
        name: "About",
        route: "about",
      },
    ],
  },
  sentry: {
    dsn: "tbd",
  },
  slug: "app",
  websiteDomain: "test",
};

export default config;
