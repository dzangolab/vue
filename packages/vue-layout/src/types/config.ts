import type { MenuItem } from "./menu";
import type { AppConfig } from "@dzangolab/vue3-config";
import type { LocaleMessages, VueMessageType } from "@dzangolab/vue3-i18n";

interface DzangolabVue3LayoutConfig {
  mainMenu?: MenuItem[];
}

interface DzangolabVue3LayoutPluginOptions {
  config: AppConfig;
  translations?: LocaleMessages<VueMessageType>;
}

export type { DzangolabVue3LayoutConfig, DzangolabVue3LayoutPluginOptions };
