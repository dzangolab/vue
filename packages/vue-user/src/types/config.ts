import { DefineComponent } from "vue";

import type { RouteOverrides } from "./router";
import type {
  IsEmailOptions,
  StrongPasswordOptions,
} from "@dzangolab/vue3-form";

interface DzangolabVueUserConfig {
  features?: {
    signUp?: {
      emailVerification?: boolean;
      termsAndConditions?: {
        display?: boolean;
        label: DefineComponent;
        showCheckbox?: boolean;
      };
    };
    updateEmail?: boolean;
  };
  password?: {
    minLength: number;
  };
  options?: {
    email?: IsEmailOptions;
    password?: StrongPasswordOptions;
  };
  routes?: RouteOverrides;
  socialLogins?: string[];
  supportedRoles?: string[];
}

declare module "@dzangolab/vue3-config" {
  export interface AppConfig {
    user?: DzangolabVueUserConfig;
  }
}

export type { DzangolabVueUserConfig };
