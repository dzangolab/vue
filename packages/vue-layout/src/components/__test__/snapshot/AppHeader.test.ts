import configPlugin from "@dzangolab/vue3-config";
import i18nPlugin from "@dzangolab/vue3-i18n";
import { mount, RouterLinkStub } from "@vue/test-utils";
import { describe, expect, it } from "vitest";

import AppHeader from "@/components/AppHeader.vue";

import config from "../config";

import type { VueWrapper } from "@vue/test-utils";

describe("AppHeader Snapshot", () => {
  it("matches snapshot", () => {
    const wrapper: VueWrapper = mount(AppHeader, {
      global: {
        plugins: [
          [
            configPlugin,
            {
              config,
            },
          ],
          [
            i18nPlugin,
            {
              config,
            },
          ],
        ],
        stubs: {
          RouterLink: RouterLinkStub,
        },
      },
    });

    expect(wrapper.element).toMatchSnapshot();
  });
});
