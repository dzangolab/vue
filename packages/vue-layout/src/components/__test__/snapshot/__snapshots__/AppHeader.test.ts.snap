// Vitest Snapshot v1

exports[`AppHeader Snapshot > matches snapshot 1`] = `
<header>
  
  <logo-stub
    class="logo"
    logo="logo.png"
  />
  
  
  <main-menu-stub
    class="mainMenu"
  />
  
  
  
  
  <locale-switcher-stub
    class="locales"
  />
  
  
  <div
    class="toggle"
  >
    <anonymous-stub
      height="1.5rem"
      icon="fa6-solid:bars"
    />
  </div>
  
</header>
`;
